### PR TITLE
Feature/nsurlsession and uploads

### PR DESCRIPTION
**Pull request by largosblade**
Providing method to add image with name
CreateImageUploadBodyData can be used with the images array to generate
the image multipart form data for uploading

Will wait to merge this in until the list below has been completed

Please feel free to add more things to the list below
- [ ] Test the file uploads
- [x] Change the way the submit function works to send all the data at once

### CODE REVIEW

File: Classes/Doorbell.m [hunk 1]
```diff
@@ -170,31 +185,83 @@ - (void)sendSubmit:(NSString*)message email:(NSString*)email
...
+    NSURLSessionConfiguration *conf = [[NSURLSessionConfiguration alloc] init];
+    [conf setHTTPAdditionalHeaders:@{@"Accept"        : @"application/json",
+                                     @"Content-Type"  : [NSString stringWithFormat:@"multipart/form-data; boundary=%@", self.imageBoundary]
+                                     }];
+    NSURLSession *uploadSession = [NSURLSession sessionWithConfiguration:conf];
 
-                                   [self manageSubmitResponse:httpResp content:content];
-                               }
-                           }];
+    NSString *query = [NSString stringWithFormat:EndpointTemplate, self.appID, @"submit", self.apiKey];
```
Comment (manavo) [inline Classes/Doorbell.m]: This will have to be the upload endpoint, right? It is just `/upload` instead of `/submit`

Comment (largosblade) [inline Classes/Doorbell.m]: Yeah I need to change that bit up.  Just wanted to get the structure of it there.

**Comment by largosblade** [comment]
screen shot 2016-01-27 at 8 50 30 am

**Comment by manavo** [comment]
Looks great at first glance! Will try and do some tests ASAP and try to get it merged! :tada:

**Comment by largosblade** [comment]
If you are good with this functionality go ahead and merge this so we can delete the branch.  I'll just make another branch/PR off master for the VC refactoring.

**Comment by manavo** [comment]
:shipit:

**Pull request merged**